subiquity/common/types.py

```diff
@@ -56,16 +56,25 @@ class ErrorReportRef:
 
 
 class ApplicationState(enum.Enum):
+    """ Represents the state of the application at a given time. """
+
+    # States reported during the initial stages of the installation
     STARTING_UP = enum.auto()
     CLOUD_INIT_WAIT = enum.auto()
     EARLY_COMMANDS = enum.auto()
+
+    # States reported during installation. This sequence should be expected
```

Comment:
Well, NEEDS_CONFIRMATION would be once only I think?

Reply:
Yep, I agree. I think I got confused while reading the spec. I'll update this comment.